File: graphql-test/src/main/java/com/junyss/graphqltest/software/model/entity/Software.java
package com.junyss.graphqltest.software.model.entity;

import javax.persistence.Entity;
import javax.persistence.Id;

import com.junyss.graphqltest.software.model.dto.request.SoftwareRequestDto;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Software {

	@Id private String softwareId;
	private String usedBy;
	private String developedBy;
	private String description;

	public static Software toEntity (SoftwareRequestDto softwareRequestDto) {
		return new Software(
			softwareRequestDto.getSoftwareId(),
			softwareRequestDto.getUsedBy(),
			softwareRequestDto.getDevelopedBy(),
			softwareRequestDto.getDescription());
	}

	public void updateUsedBy(String usedBy) {
		this.usedBy = usedBy;
	}

	public void updateDevelopedBy(String developedBy) {
		this.developedBy = developedBy;
	}

	public void updateDescription(String description) {
		this.description = description;
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/software/model/entity/Software.kt
package com.junyharangstudy.kotlingraphqltest.api.software.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id

@Entity
class Software (
    @Id var softwareId: String,
    var usedBy: String,
    var developedBy: String,
    var description: String?
) {
    fun updateUsedBy(usedBy: String) {
        this.usedBy = usedBy
    }

    fun updateDevelopedBy(developedBy: String) {
        this.developedBy = developedBy
    }

    fun updateDescription(description: String) {
        this.description = description
    }
}